elf_reader.go

```diff
@@ -413,44 +426,69 @@ func (ec *elfCode) relocateInstruction(ins *asm.Instruction, rel elf.Symbol) err
 		}
 
 	case programSection:
-		if ins.OpCode.JumpOp() != asm.Call {
-			return fmt.Errorf("not a call instruction: %v", ins)
+		if ins.OpCode.JumpOp() != asm.Call && !ins.OpCode.IsDWordLoad() {
+			return fmt.Errorf("neither a call nor a load instruction: %v", ins)
 		}
 
-		if ins.Src != asm.PseudoCall {
+		if ins.OpCode.JumpOp() == asm.Call && ins.Src != asm.PseudoCall {
 			return fmt.Errorf("call: %s: incorrect source register", name)
 		}
 
 		switch typ {
-		case elf.STT_NOTYPE, elf.STT_FUNC:
+		case elf.STT_NOTYPE:
 			if bind != elf.STB_GLOBAL {
 				return fmt.Errorf("call: %s: unsupported binding: %s", name, bind)
 			}
+		case elf.STT_FUNC:
+			if bind != elf.STB_GLOBAL {
+				return fmt.Errorf("load/call: %s: unsupported binding: %s", name, bind)
+			}
+
+			if ins.OpCode.IsDWordLoad() {
```

Comment:
So it's not valid for a call to reference STT_FUNC, but it's ok for a load? If that is true we should return an error for calls rather than doing nothing.